### ISSUE TITLE
npm -v prints incorrect version npm is 5.8, npm -v prints 5.6

### ISSUE THREAD

**Issue by drmason13** [npm/npm#20339]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [x] npm is producing incorrect or undesirable behavior.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?
```
$ npm install npm@latest -g
/home/masond/.npm-global/bin/npm -> /home/masond/.npm-global/lib/node_modules/npm/bin/npm-cli.js
/home/masond/.npm-global/bin/npx -> /home/masond/.npm-global/lib/node_modules/npm/bin/npx-cli.js
+ npm@5.8.0
added 523 packages in 4.303s
$ npm -v
5.6.0
$ npm list --global --depth 0/home/masond/.npm-global/lib
├── custom-react-scripts@0.2.2
├── firebase-tools@3.17.7
└── npm@5.8.0

npm ERR! peer dep missing: ajv@^6.0.0, required by ajv-keywords@3.1.0
$ npm -v
5.6.0
```
#### How can the CLI team reproduce the problem?

no npm-debug.log (it isn't created)
here is the verbose output of install instead (not the first time, hence the re-install?)
```
$ npm install npm@latest -g --verbose
npm info it worked if it ends with ok
npm verb cli [ '/usr/local/bin/node',
npm verb cli   '/usr/local/bin/npm',
npm verb cli   'install',
npm verb cli   'npm@latest',
npm verb cli   '-g',
npm verb cli   '--verbose' ]
npm info using npm@5.6.0
npm info using node@v9.2.0
npm verb npm-session 34174ee51118c12e
npm http fetch GET 304 https://registry.npmjs.org/npm 256ms (from cache)
npm verb correctMkdir /home/masond/.npm/_locks correctMkdir not in flight; initializing
npm verb lock using /home/masond/.npm/_locks/staging-384c763097226e4e.lock for /home/masond/.npm-global/lib/node_modules/.staging
npm info lifecycle npm@5.8.0~preuninstall: npm@5.8.0
npm info lifecycle npm@5.8.0~uninstall: npm@5.8.0
npm verb unbuild rmStuff npm@5.8.0 from /home/masond/.npm-global/lib/node_modules
npm verb rmMans man files are [ '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-access.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-adduser.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-bin.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-bugs.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-build.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-bundle.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-cache.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-ci.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-completion.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-config.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-dedupe.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-deprecate.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-dist-tag.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-docs.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-doctor.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-edit.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-explore.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-help-search.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-help.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-init.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-install-test.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-install.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-link.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-logout.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-ls.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-outdated.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-owner.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-pack.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-ping.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-prefix.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-profile.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-prune.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-publish.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-README.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-rebuild.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-repo.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-restart.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-root.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-run-script.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-search.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-shrinkwrap.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-star.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-stars.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-start.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-stop.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-team.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-test.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-token.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-uninstall.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-unpublish.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-update.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-version.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-view.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-whoami.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npx.1',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man5/npm-folders.5',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man5/npm-global.5',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man5/npm-json.5',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man5/npm-package-locks.5',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man5/npm-shrinkwrap.json.5',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man5/npmrc.5',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man5/package-lock.json.5',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man5/package.json.5',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man7/npm-coding-style.7',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man7/npm-config.7',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man7/npm-developers.7',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man7/npm-disputes.7',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man7/npm-index.7',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man7/npm-orgs.7',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man7/npm-registry.7',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man7/npm-scope.7',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man7/npm-scripts.7',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man7/removing-npm.7',
npm verb rmMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man7/semver.7' ] in /home/masond/.npm-global/share/man
npm info lifecycle npm@5.8.0~postuninstall: npm@5.8.0
npm info lifecycle npm@5.8.0~preinstall: npm@5.8.0
npm info linkStuff npm@5.8.0
npm verb linkBins npm@5.8.0
npm verb linkBins [ { npm: './bin/npm-cli.js', npx: './bin/npx-cli.js' },
npm verb linkBins   '/home/masond/.npm-global/bin',
npm verb linkBins   true ]
npm verb linkMans npm@5.8.0
npm verb linkMans man files are [ '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-access.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-adduser.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-bin.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-bugs.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-build.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-bundle.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-cache.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-ci.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-completion.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-config.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-dedupe.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-deprecate.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-dist-tag.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-docs.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-doctor.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-edit.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-explore.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-help-search.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-help.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-init.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-install-test.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-install.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-link.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-logout.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-ls.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-outdated.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-owner.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-pack.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-ping.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-prefix.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-profile.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-prune.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-publish.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-README.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-rebuild.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-repo.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-restart.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-root.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-run-script.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-search.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-shrinkwrap.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-star.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-stars.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-start.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-stop.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-team.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-test.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-token.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-uninstall.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-unpublish.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-update.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-version.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-view.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm-whoami.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npm.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man1/npx.1',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man5/npm-folders.5',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man5/npm-global.5',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man5/npm-json.5',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man5/npm-package-locks.5',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man5/npm-shrinkwrap.json.5',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man5/npmrc.5',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man5/package-lock.json.5',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man5/package.json.5',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man7/npm-coding-style.7',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man7/npm-config.7',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man7/npm-developers.7',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man7/npm-disputes.7',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man7/npm-index.7',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man7/npm-orgs.7',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man7/npm-registry.7',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man7/npm-scope.7',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man7/npm-scripts.7',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man7/removing-npm.7',
npm verb linkMans   '/home/masond/.npm-global/lib/node_modules/npm/man/man7/semver.7' ] in /home/masond/.npm-global/share/man
/home/masond/.npm-global/bin/npm -> /home/masond/.npm-global/lib/node_modules/npm/bin/npm-cli.js
/home/masond/.npm-global/bin/npx -> /home/masond/.npm-global/lib/node_modules/npm/bin/npx-cli.js
npm info lifecycle npm@5.8.0~install: npm@5.8.0
npm info lifecycle npm@5.8.0~postinstall: npm@5.8.0
npm verb unlock done using /home/masond/.npm/_locks/staging-384c763097226e4e.lock for /home/masond/.npm-global/lib/node_modules/.staging
+ npm@5.8.0
updated 1 package in 3.927s
npm verb exit [ 0, true ]
npm info ok 
$ npm -v
5.6.0
```
### supporting information:

 - `npm -v` prints:
```
5.6
```
 - `node -v` prints:
```
v9.2.0
```
 - `npm config get registry` prints:
```
https://registry.npmjs.org/
```
 - Windows, OS X/macOS, or Linux?:
 - Network issues:
   - Geographic location where npm was run:
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by kenany** [comment]
You have two installation of `npm`:

  - `/usr/local/bin/npm`
  - `/home/masond/.npm-global/bin/npm`

**Comment by drmason13** [comment]
Oh gosh! That is embarrassing. Thanks for pointing it out for me.

I had made a change recently to allow installing with npm without sudo which uses the home folder to install to, following https://docs.npmjs.com/getting-started/fixing-npm-permissions (option two).

Removing the /usr/local/bin/npm fixed!

```
$ whereis npm
npm: /usr/local/bin/npm /home/masond/.npm-global/bin/npm
$ cd /usr/local/bin
/usr/local/bin$ sudo mv npm not-npm
/usr/local/bin$ npm -v
5.6.0
/usr/local/bin$ npm install --global npm@latest
/home/masond/.npm-global/bin/npm -> /home/masond/.npm-global/lib/node_modules/npm/bin/npm-cli.js
/home/masond/.npm-global/bin/npx -> /home/masond/.npm-global/lib/node_modules/npm/bin/npx-cli.js
+ npm@5.8.0
added 89 packages, removed 42 packages and updated 67 packages in 4.562s


   ╭─────────────────────────────────────╮
   │                                     │
   │   Update available 5.6.0 → 5.8.0    │
   │       Run npm i npm to update       │
   │                                     │
   ╰─────────────────────────────────────╯

/usr/local/bin$ npm -v
5.8.0
/usr/local/bin$ whereis npm
npm: /home/masond/.npm-global/bin/npm
```
